README.md

```diff
@@ -6,76 +6,86 @@
 [![GitHub release](https://img.shields.io/github/release/kubernetes/dashboard.svg)](https://github.com/kubernetes/dashboard/releases/latest)
 [![Greenkeeper badge](https://badges.greenkeeper.io/kubernetes/dashboard.svg)](https://greenkeeper.io/)
 
-Kubernetes Dashboard is a general purpose, web-based UI for Kubernetes clusters. It allows users to
-manage applications running in the cluster and troubleshoot them, as well as manage the cluster
-itself.
+Kubernetes Dashboard is a general purpose, web-based UI for Kubernetes clusters. It allows users to manage applications
+running in the cluster and troubleshoot them, as well as manage the cluster itself.
 
 ![Dashboard UI workloads page](docs/dashboard-ui.png)
 
 ## Deployment
-It is likely that the Dashboard is already installed on your cluster. Check with the following command:
-```shell
-$ kubectl get pods --all-namespaces | grep dashboard
-```
 
-If it is missing, you can install the latest stable release by running the following command:
-```shell
-$ kubectl create -f https://git.io/kube-dashboard
+**IMPORTANT:** As of version 1.7 Dashboard uses minimal privileges setup by default. It means, that some manual steps
+are required to make it fully secure and functional. Before performing any steps read [Access Control](
+https://github.com/kubernetes/dashboard/wiki/Access-control) guide.
+
+### Recommended setup
+
+Full security can be ensured only by accessing Dashboard over HTTPS. In order to enable HTTPS mode certificates need
+to be passed to the application. This can be achieved by providing `--tls-cert-file` and `--tls-cert-key` flags to the
```

Comment:
Can you please elaborate why we recomend to configure HTTPS? Isn't `kubectl proxy` sufficient? Do we want to protect the traffic inside the cluster?

Reply:
> Am I supposed to modify the deployment yaml? Couldn't it be detected in case the config map is mounted?

User is only supposed to follow described steps. Nothing else. Only manual step is to generate certificates that will be used by Dashboard to enable HTTPS. They will be used to create configmap (we'll change it to secret). This way certs can be mounted into Dashboard container and used to serve HTTPS.

> Can you please elaborate why we recomend to configure HTTPS? 

Full security can be ensured only by accessing Dashboard over HTTPS.

This is the main reason and in case user wants to know more he can read our wiki.

> Isn't kubectl proxy sufficient? Do we want to protect the traffic inside the cluster?

It's Dashboard that has to ensure security, not people accessing it by using some secure tunnel. In order to even see login view Dashboard has to be accessed over HTTPS. This does not mean that it can be accessed through secure tunnel. User has to use HTTPS endpoints exposed by Dashboard. User credentials can never be sent with plain text between frontend and backend.

All that is described in detail in our wiki guides.

Reply:
> User is only supposed to follow described steps. Nothing else. Only manual step is to generate certificates that will be used by Dashboard to enable HTTPS.

Ok fine. But text is misleading.

> It's Dashboard that has to ensure security, not people accessing it by using some secure tunnel. In order to even see login view Dashboard has to be accessed over HTTPS. This does not mean that it can be accessed through secure tunnel. User has to use HTTPS endpoints exposed by Dashboard. User credentials can never be sent with plain text between frontend and backend.

If I understand you correctly you want to prefent the user from misconfiguration. Getting certificates from lets-encrypt or distributing the self-generated root certificate to all clients is some effort. 1. It is effort that (maybe) is not required when accessing through secure tunnel and 2. insecure if not done properly (of course the browser will warn). So, I wonder if making it optional would be better choice

Reply:
> Ok fine. But text is misleading.

Ok, it might be. We can change "This can be achieved by providing --tls-cert-file and --tls-cert-key flags to the Dashboard." sentence and add short explanation that this is already done and user has to just follow the instructions.

> If I understand you correctly you want to prefent the user from misconfiguration. Getting certificates from lets-encrypt or distributing the self-generated root certificate to all clients is some effort. 1. It is effort that (maybe) is not required when accessing through secure tunnel and 2. insecure if not done properly (of course the browser will warn). So, I wonder if making it optional would be better choice

No. Think of Dashboard like any other regular webapp. Do not think about kubernetes for a second. You are a user that opens a page in his browser. Would you rather pass your credentials to HTTP page and send them with plain text to the server or use HTTPS endpoint and know that traffic will be encrypted?

Dashboard (and any other web page) does not care if user creates some additional tunnel to access it. It has to expose HTTPS endpoints itself. This is the only way to make sure that traffic between frontend and backend is encrypted and not easily sniffed. If user is indeed accessing HTTPS endpoints then and only then we are enabling login view. Otherwise the only way of acting as an user is to pass and use `Authorization: Bearer <token>` feature.